ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -319,12 +319,25 @@ private PutOperation(RouterConfig routerConfig, NonBlockingRouterMetrics routerM
     bytesFilledSoFar = 0;
     chunkCounter = -1;
     putChunks = new ConcurrentLinkedQueue<>();
-    reservedMetadataIdMetrics = ReservedMetadataIdMetrics.getReservedMetadataIdMetrics(routerMetrics.getMetricRegistry());
-    metadataPutChunk = new MetadataPutChunk();
+    reservedMetadataIdMetrics =
+        ReservedMetadataIdMetrics.getReservedMetadataIdMetrics(routerMetrics.getMetricRegistry());
     chunkFillerChannel = new ByteBufferAsyncWritableChannel(writableChannelEventListener);
     isEncryptionEnabled = passedInBlobProperties.isEncrypted();
     restRequest = options.getRestRequest();
     loggingContext = makeLoggingContext();
+    if (chunksToStitch.isEmpty()) {
```

Comment:
chunksToStitch can be null here.  Should be

```
if (chunksToStitch !=null && chunksToStitch.isEmpty())
```

Suppose we should have unit test fail. If not, we should add unit test.